code/modules/modular_computers/file_system/programs/civilian/cargo_control.dm

```diff
@@ -0,0 +1,191 @@
+	data["page"] = page
+	//Send teh status message
+	data["status_message"] = status_message
+
+	//Pass the ID Data
+	var/obj/item/weapon/card/id/user_id_card = user.GetIdCard()
+	last_user_name = GetNameAndAssignmentFromId(user_id_card)
+	data["username"] = last_user_name
+
+
+	//TODO: Send the data along depending on the page we are on
+
+	var/list/submitted_orders = SScargo.get_orders_by_status("submitted",1)
+	data["order_submitted_number"] = submitted_orders.len
+	data["order_submitted_value"] = SScargo.get_orders_value_by_status("submitted",1)
+	data["order_submitted_suppliers"] = SScargo.get_order_suppliers_by_status("submitted",1)
+	data["order_submitted_shuttle_time"] = SScargo.get_pending_shipment_time("submitted")
+	data["order_submitted_shuttle_price"] = SScargo.get_pending_shipment_cost("submitted")
+	if(page == "overview_submitted")
+		data["order_list"] = submitted_orders
+
+	var/list/approved_orders = SScargo.get_orders_by_status("approved",1)
+	data["order_approved_number"] = approved_orders.len
+	data["order_approved_value"] = SScargo.get_orders_value_by_status("approved",1)
+	data["order_approved_suppliers"] = SScargo.get_order_suppliers_by_status("approved",1)
+	data["order_approved_shuttle_time"] = SScargo.get_pending_shipment_time("approved")
+	data["order_approved_shuttle_price"] = SScargo.get_pending_shipment_cost("approved")
+	if(page == "overview_approved")
+		data["order_list"] = approved_orders
+
+	var/list/shipped_orders = SScargo.get_orders_by_status("shipped",1)
+	data["order_shipped_number"] = shipped_orders.len
+	data["order_shipped_value"] = SScargo.get_orders_value_by_status("shipped",1)
+	if(page == "overview_shipped")
+		data["order_list"] = shipped_orders
+
+
+	if(page == "order_details")
+		data["order_details"] = order_details
+
+
+	data["cargo_money"] = SScargo.get_cargo_money()
+	data["handling_fee"] = SScargo.get_handlingfee()
+
+	//Shuttle Stuff
+	var/datum/shuttle/ferry/supply/shuttle = SScargo.shuttle
+	if(shuttle)
+		data["shuttle_available"] = 1
+		data["shuttle_has_arrive_time"] = shuttle.has_arrive_time()
+		data["shuttle_eta_minutes"] = shuttle.eta_minutes()
+		data["shuttle_can_launch"] = shuttle.can_launch()
+		data["shuttle_can_cancel"] = shuttle.can_cancel()
+		data["shuttle_can_force"] = shuttle.can_force()
+		data["shuttle_at_station"] = shuttle.at_station()
+		if(shuttle.docking_controller)
+			data["shuttle_docking_status"] = shuttle.docking_controller.get_docking_status()
+		else
+			data["shuttle_docking_status"] = "error"
+	else
+		data["shuttle_available"] = 0
+
+	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "cargo_control.tmpl", name, 850, 600, state = state)
+		ui.auto_update_layout = 1
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
+
+/datum/nano_module/program/civilian/cargocontrol/Topic(href, href_list)
+	//TODO: Implement return message to notify user of whats going on
+
+	if(!SScargo)
```

Comment:
The old system had it as a holdover from when this used a process, it should be fine to remove as the only time this might ever be null is if something calls `del()` (not qdel) on SScargo.